src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sinks/cassandra/CassandraEventsSink.scala

```diff
@@ -0,0 +1,133 @@
+package com.microsoft.partnercatalyst.fortis.spark.sinks.cassandra
+
+import java.util.UUID
+import com.datastax.spark.connector.writer.{RowWriterFactory, WriteConf}
+import com.microsoft.partnercatalyst.fortis.spark.dto.FortisEvent
+import org.apache.spark.sql.{Dataset, SaveMode, SparkSession}
+import org.apache.spark.streaming.dstream.DStream
+import com.datastax.spark.connector._
+import com.datastax.spark.connector.mapper.ColumnMapper
+import com.microsoft.partnercatalyst.fortis.spark.sinks.cassandra.aggregators._
+import com.microsoft.partnercatalyst.fortis.spark.sinks.cassandra.dto._
+import org.apache.spark.rdd.RDD
+import com.microsoft.partnercatalyst.fortis.spark.sinks.cassandra.udfs._
+import org.apache.spark.streaming.Time
+
+import scala.util.{Failure, Success, Try}
+
+object CassandraEventsSink{
+  private val KeyspaceName = "fortis"
+  private val TableEvent = "events"
+  private val TableEventTopics = "eventtopics"
+  private val TableEventPlaces = "eventplaces"
+  private val TableEventBatches = "eventbatches"
+  private val CassandraFormat = "org.apache.spark.sql.cassandra"
+  private val CassandraRetryIntervalMS = 2000
+  private val CassandraMaxRetryAttempts = 3
+
+  def apply(dstream: DStream[FortisEvent], sparkSession: SparkSession): Unit = {
+      val batchStream = dstream.foreachRDD{ (eventsRDD, time: Time) => {
+        if(!eventsRDD.isEmpty) {
+          val batchid = UUID.randomUUID().toString
+          val fortisEvents = writeFortisEvents(eventsRDD, batchid)
+          val aggregators = Seq(new PopularPlacesAggregator, new PopularTopicAggregator).par
+          val session = SparkSession.builder().config(eventsRDD.sparkContext.getConf)
+            .appName(eventsRDD.sparkContext.appName)
+            .getOrCreate()
+
+          if(session != null){
```

Comment:
If the connection is dropped, which should never happen. I can remove the check is it should never happen.